src/Symfony/Bundle/FrameworkBundle/Controller/RedirectController.php

```diff
@@ -23,10 +23,37 @@
  * Redirects a request to another URL.
  *
  * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @final since version 3.4
  */
 class RedirectController implements ContainerAwareInterface
 {
-    use ContainerAwareTrait;
+    /**
+     * @deprecated since version 3.4, to be removed in 4.0
+     */
+    protected $container;
+
+    private $router;
+    private $httpPort;
+    private $httpsPort;
+
+    public function __construct(UrlGeneratorInterface $router = null, $httpPort = null, $httpsPort = null)
+    {
+        $this->router = $router;
+        $this->httpPort = $httpPort;
+        $this->httpsPort = $httpsPort;
+    }
+
+    /**
+     * @deprecated since version 3.4, to be removed in 4.0 alongside with the ContainerAwareInterface type.
+     */
+    public function setContainer(ContainerInterface $container = null)
+    {
+        @trigger_error(sprintf('The "%s()" method is deprecated since version 3.4 and will be removed in 4.0. Inject a PSR-11 container using the constructor instead.', __METHOD__), E_USER_DEPRECATED);
```

Comment:
huh? the constructor does not accept a container.

Reply:
fixed